### PR TITLE
Allow to observe feature flags

### PR DESCRIPTION
**Pull request by MiSikora**
## Description

This PR allows to observe feature flags via `Flow`. I need it to properly support ads feature detection in the player as I'm running into some edge cases which block my progress without doing a lot of work.

## Testing Instructions

1. Apply [this patch](https://github.com/user-attachments/files/20606727/firabase-test.patch).
```
curl -L https://github.com/user-attachments/files/20606727/firabase-test.patch | git apply
```
2. Install `debugProd` variant.
3. Open Logcat and filter by `Testing` tag.
4. Notice `TEST_FREE_FEATURE enabled: true` log.
5. Go to the Firebase Remote config and modify `test_free_feature` key. Publish the changes.
6. Wait 10 seconds.
7. You should see`TEST_FREE_FEATURE enabled: <value>` in the Logcat.
8. Repeat steps 5-7 a couple of times to verify updates.

## Checklist
- [ ] ~If this is a user-facing change, I have added an entry in CHANGELOG.md~
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [ ] ~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~
- [ ] ~Any jetpack compose components I added or changed are covered by compose previews~
- [ ] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/featureflag/Feature.kt [hunk 1]
```diff
@@ -133,31 +133,31 @@ enum class Feature(
 
     // This is set of features used only for testing purposes.
     TEST_FREE_FEATURE(
-        key = "test_free_feautre",
+        key = "test_free_feature",
```
Comment (sztomek) [inline modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/featureflag/Feature.kt:136]: nice 😅 long-lived easter eggs :D

**Review by sztomek — APPROVED** [review]

**Comment by sztomek** [comment]
~Please note that I wasn't able to test it properly as I'm not added to the Firebase project yet... :| ~

**update**
after Michal has added me to Firebase, I was finally able to verify that things work as expected
Screenshot 2025-06-05 at 12 06 47
